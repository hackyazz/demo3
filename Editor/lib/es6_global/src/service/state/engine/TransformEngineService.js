

import * as TransformAPI$Wonderjs from "../../../../../../node_modules/wonder.js/lib/es6_global/src/api/TransformAPI.js";
import * as HierachyTransformService$Wonderjs from "../../../../../../node_modules/wonder.js/lib/es6_global/src/service/record/main/transform/HierachyTransformService.js";
import * as RecordTransformMainService$Wonderjs from "../../../../../../node_modules/wonder.js/lib/es6_global/src/service/state/main/transform/RecordTransformMainService.js";

function setLocalPosition(localPosition, transform, engineState) {
  return TransformAPI$Wonderjs.setTransformLocalPosition(transform, localPosition, engineState);
}

function setTransformLocalEulerAngles(localEulerAngles, transform, engineState) {
  return TransformAPI$Wonderjs.setTransformLocalEulerAngles(transform, localEulerAngles, engineState);
}

function getParent(child, state) {
  return HierachyTransformService$Wonderjs.getParent(child, RecordTransformMainService$Wonderjs.getRecord(state));
}

var setParent = TransformAPI$Wonderjs.setTransformParent;

var setParentKeepOrder = TransformAPI$Wonderjs.setTransformParentKeepOrder;

var getChildren = TransformAPI$Wonderjs.unsafeGetTransformChildren;

function setLocalEulerAngles(value, component, engineState) {
  return TransformAPI$Wonderjs.setTransformLocalEulerAngles(component, value, engineState);
}

function setLocalScale(value, component, engineState) {
  return TransformAPI$Wonderjs.setTransformLocalScale(component, value, engineState);
}

var getLocalToWorldMatrixTypeArray = TransformAPI$Wonderjs.getTransformLocalToWorldMatrixTypeArray;

var getPosition = TransformAPI$Wonderjs.getTransformPosition;

var setPosition = TransformAPI$Wonderjs.setTransformPosition;

var getLocalPosition = TransformAPI$Wonderjs.getTransformLocalPosition;

var changeChildOrder = TransformAPI$Wonderjs.changeChildOrder;

var getGameObjectByTransform = TransformAPI$Wonderjs.unsafeGetTransformGameObject;

var getLocalRotation = TransformAPI$Wonderjs.getTransformLocalRotation;

var setLocalRotation = TransformAPI$Wonderjs.setTransformLocalRotation;

var getRotation = TransformAPI$Wonderjs.getTransformRotation;

var setRotation = TransformAPI$Wonderjs.setTransformRotation;

var getLocalEulerAngles = TransformAPI$Wonderjs.getTransformLocalEulerAngles;

var getEulerAngles = TransformAPI$Wonderjs.getTransformEulerAngles;

var setEulerAngles = TransformAPI$Wonderjs.setTransformEulerAngles;

var getLocalScale = TransformAPI$Wonderjs.getTransformLocalScale;

var getScale = TransformAPI$Wonderjs.getTransformScale;

var setScale = TransformAPI$Wonderjs.setTransformScale;

var lookAt = TransformAPI$Wonderjs.lookAt;

var rotateLocalOnAxis = TransformAPI$Wonderjs.rotateLocalOnAxis;

var rotateWorldOnAxis = TransformAPI$Wonderjs.rotateWorldOnAxis;

export {
  getPosition ,
  setPosition ,
  getLocalPosition ,
  setLocalPosition ,
  setTransformLocalEulerAngles ,
  getParent ,
  setParent ,
  setParentKeepOrder ,
  getChildren ,
  changeChildOrder ,
  getGameObjectByTransform ,
  getLocalRotation ,
  setLocalRotation ,
  getRotation ,
  setRotation ,
  getLocalEulerAngles ,
  setLocalEulerAngles ,
  getEulerAngles ,
  setEulerAngles ,
  getLocalScale ,
  setLocalScale ,
  getScale ,
  setScale ,
  getLocalToWorldMatrixTypeArray ,
  lookAt ,
  rotateLocalOnAxis ,
  rotateWorldOnAxis ,
  
}
/* TransformAPI-Wonderjs Not a pure module */
