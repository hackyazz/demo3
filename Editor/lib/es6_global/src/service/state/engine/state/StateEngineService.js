

import * as StateAPI$Wonderjs from "../../../../../../../node_modules/wonder.js/lib/es6_global/src/api/StateAPI.js";

var deepCopyForRestore = StateAPI$Wonderjs.deepCopyForRestore;

var createState = StateAPI$Wonderjs.createState;

var createStateData = StateAPI$Wonderjs.createStateData;

var getStateFromData = StateAPI$Wonderjs.getStateFromData;

var unsafeGetState = StateAPI$Wonderjs.unsafeGetState;

var setStateToData = StateAPI$Wonderjs.setStateToData;

var setState = StateAPI$Wonderjs.setState;

var setIsDebug = StateAPI$Wonderjs.setIsDebug;

export {
  deepCopyForRestore ,
  createState ,
  createStateData ,
  getStateFromData ,
  unsafeGetState ,
  setStateToData ,
  setState ,
  setIsDebug ,
  
}
/* StateAPI-Wonderjs Not a pure module */
