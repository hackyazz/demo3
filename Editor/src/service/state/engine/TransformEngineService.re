/* open Wonderjs; */

/* open TransformType; */

let getPosition = Wonderjs.TransformAPI.getTransformPosition;

let setPosition = Wonderjs.TransformAPI.setTransformPosition;

let getLocalPosition = Wonderjs.TransformAPI.getTransformLocalPosition;

let setLocalPosition = (localPosition, transform, engineState) =>
  Wonderjs.TransformAPI.setTransformLocalPosition(
    transform,
    localPosition,
    engineState,
  );

let setTransformLocalEulerAngles = (localEulerAngles, transform, engineState) =>
  Wonderjs.TransformAPI.setTransformLocalEulerAngles(
    transform,
    localEulerAngles,
    engineState,
  );

let getParent = (child: Wonderjs.TransformType.transform, state) =>
  Wonderjs.HierachyTransformService.getParent(
    child,
    Wonderjs.RecordTransformMainService.getRecord(state),
  );

let setParent =
    (
      parent: Wonderjs.TransformType.transform,
      child: Wonderjs.TransformType.transform,
      state,
    ) =>
  Wonderjs.TransformAPI.setTransformParent(
    Js.Nullable.return(parent),
    child,
    state,
  );

let setParentKeepOrder =
    (
      parent: Wonderjs.TransformType.transform,
      child: Wonderjs.TransformType.transform,
      state,
    ) =>
  Wonderjs.TransformAPI.setTransformParentKeepOrder(
    Js.Nullable.return(parent),
    child,
    state,
  );

let getChildren = (transform: Wonderjs.TransformType.transform, state) =>
  Wonderjs.TransformAPI.unsafeGetTransformChildren(transform, state);

let changeChildOrder = Wonderjs.TransformAPI.changeChildOrder;

let getGameObjectByTransform = Wonderjs.TransformAPI.unsafeGetTransformGameObject;

let getLocalRotation = Wonderjs.TransformAPI.getTransformLocalRotation;

let setLocalRotation = Wonderjs.TransformAPI.setTransformLocalRotation;

let getRotation = Wonderjs.TransformAPI.getTransformRotation;

let setRotation = Wonderjs.TransformAPI.setTransformRotation;

let getLocalEulerAngles = Wonderjs.TransformAPI.getTransformLocalEulerAngles;

let setLocalEulerAngles = (value, component, engineState) =>
  Wonderjs.TransformAPI.setTransformLocalEulerAngles(
    component,
    value,
    engineState,
  );

let getEulerAngles = Wonderjs.TransformAPI.getTransformEulerAngles;

let setEulerAngles = Wonderjs.TransformAPI.setTransformEulerAngles;

let getLocalScale = Wonderjs.TransformAPI.getTransformLocalScale;

let setLocalScale = (value, component, engineState) =>
  Wonderjs.TransformAPI.setTransformLocalScale(component, value, engineState);

let getScale = Wonderjs.TransformAPI.getTransformScale;

let setScale = Wonderjs.TransformAPI.setTransformScale;

let getLocalToWorldMatrixTypeArray = (transform, engineState) =>
  Wonderjs.TransformAPI.getTransformLocalToWorldMatrixTypeArray(
    transform,
    engineState,
  );

let lookAt = Wonderjs.TransformAPI.lookAt;

let rotateLocalOnAxis = Wonderjs.TransformAPI.rotateLocalOnAxis;

let rotateWorldOnAxis = Wonderjs.TransformAPI.rotateWorldOnAxis;