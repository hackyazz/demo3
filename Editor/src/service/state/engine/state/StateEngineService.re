let deepCopyForRestore = Wonderjs.StateAPI.deepCopyForRestore;

let createState = Wonderjs.StateAPI.createState;

let createStateData = Wonderjs.StateAPI.createStateData;

let getStateFromData = Wonderjs.StateAPI.getStateFromData;

let unsafeGetState = Wonderjs.StateAPI.unsafeGetState;

let setStateToData = Wonderjs.StateAPI.setStateToData;

let setState = Wonderjs.StateAPI.setState;

let setIsDebug = Wonderjs.StateAPI.setIsDebug;