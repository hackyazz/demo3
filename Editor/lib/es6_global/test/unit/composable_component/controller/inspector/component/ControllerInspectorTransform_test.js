

import * as Curry from "../../../../../../../../node_modules/bs-platform/lib/es6/curry.js";
import * as Sinon from "../../../../../../../../node_modules/wonder-bs-sinon/lib/es6_global/src/sinon.js";
import * as Sinon$1 from "sinon";
import * as Wonder_jest from "../../../../../../../../node_modules/wonder-bs-jest/lib/es6_global/src/wonder_jest.js";
import * as TestTool$WonderEditor from "../../../../../tool/TestTool.js";
import * as ReactTestTool$WonderEditor from "../../../../../tool/ReactTestTool.js";
import * as ControllerTool$WonderEditor from "../../tool/ControllerTool.js";
import * as GameObjectTool$WonderEditor from "../../../../../tool/GameObjectTool.js";
import * as BuildComponentTool$WonderEditor from "../../../../../tool/BuildComponentTool.js";
import * as StateEngineService$WonderEditor from "../../../../../../src/service/state/engine/state/StateEngineService.js";
import * as MainEditorSceneTool$WonderEditor from "../../../../../tool/MainEditorSceneTool.js";
import * as TransformEngineService$WonderEditor from "../../../../../../src/service/state/engine/TransformEngineService.js";
import * as MainEditorTransformTool$WonderEditor from "../../../../../integration/inspector/composable_component/sceneTree_inspector/transform/tool/MainEditorTransformTool.js";

describe("controller inspector transform", (function () {
        var sandbox = Sinon.getSandboxDefaultVal(/* () */0);
        beforeEach((function () {
                TestTool$WonderEditor.closeContractCheck(/* () */0);
                sandbox[0] = Sinon$1.sandbox.create();
                MainEditorSceneTool$WonderEditor.initState(sandbox, undefined, undefined, /* () */0);
                MainEditorSceneTool$WonderEditor.createDefaultScene(sandbox, MainEditorSceneTool$WonderEditor.setFirstCubeToBeCurrentSceneTreeNode);
                Curry._1(ControllerTool$WonderEditor.stubRequestAnimationFrame, Sinon.createEmptyStubWithJsObjSandbox(sandbox));
                return Curry._1(ControllerTool$WonderEditor.stubCancelAnimationFrame, Sinon.createEmptyStubWithJsObjSandbox(sandbox));
              }));
        afterEach((function () {
                Curry._1(Sinon.restoreSandbox, sandbox[0]);
                return TestTool$WonderEditor.openContractCheck(/* () */0);
              }));
        describe("test set transform in engine state", (function () {
                return Wonder_jest.test("current gameObject's tranform position should set into engine", (function () {
                              ControllerTool$WonderEditor.run(/* () */0);
                              var currentGameObjectTransform = GameObjectTool$WonderEditor.getCurrentSceneTreeNodeTransform(/* () */0);
                              MainEditorTransformTool$WonderEditor.changePositionXAndBlur(155, currentGameObjectTransform, undefined, undefined, /* () */0);
                              return Curry._2(Wonder_jest.Expect[/* Operators */23][/* = */5], Wonder_jest.Expect[/* expect */0](TransformEngineService$WonderEditor.getLocalPosition(currentGameObjectTransform, StateEngineService$WonderEditor.unsafeGetState(/* () */0))), /* tuple */[
                                          155,
                                          0,
                                          0
                                        ]);
                            }));
              }));
        describe("fix bug", (function () {
                return Wonder_jest.test("should refresh transform when stop", (function () {
                              ControllerTool$WonderEditor.run(/* () */0);
                              var currentGameObjectTransform = GameObjectTool$WonderEditor.getCurrentSceneTreeNodeTransform(/* () */0);
                              MainEditorTransformTool$WonderEditor.changePositionXAndBlur(155, currentGameObjectTransform, undefined, undefined, /* () */0);
                              ControllerTool$WonderEditor.stop(/* () */0);
                              return ReactTestTool$WonderEditor.createSnapshotAndMatch(BuildComponentTool$WonderEditor.buildMainEditorTransformComponent(TestTool$WonderEditor.buildEmptyAppState(/* () */0), currentGameObjectTransform));
                            }));
              }));
        return /* () */0;
      }));

export {
  
}
/*  Not a pure module */
