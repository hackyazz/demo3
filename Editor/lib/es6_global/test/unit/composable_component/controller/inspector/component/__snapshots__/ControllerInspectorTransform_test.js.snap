// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`controller inspector transform fix bug should refresh transform when stop 1`] = `
<article
  className="wonder-inspector-transform"
>
  <article
    className="inspector-item wonder-three-float-input"
  >
    <div
      className="item-header"
    >
      Position
    </div>
    <div
      className="item-content"
    >
      <article
        className="inspector-item wonder-float-input"
      >
        <div
          className="item-header component-label"
          onMouseDown={[Function]}
          onMouseMove={[Function]}
          onMouseUp={[Function]}
        >
          X
        </div>
        <div
          className="item-content"
        >
          <input
            className="input-component float-input"
            onBlur={[Function]}
            onChange={[Function]}
            type="text"
            value="0"
          />
        </div>
      </article>
      <article
        className="inspector-item wonder-float-input"
      >
        <div
          className="item-header component-label"
          onMouseDown={[Function]}
          onMouseMove={[Function]}
          onMouseUp={[Function]}
        >
          Y
        </div>
        <div
          className="item-content"
        >
          <input
            className="input-component float-input"
            onBlur={[Function]}
            onChange={[Function]}
            type="text"
            value="0"
          />
        </div>
      </article>
      <article
        className="inspector-item wonder-float-input"
      >
        <div
          className="item-header component-label"
          onMouseDown={[Function]}
          onMouseMove={[Function]}
          onMouseUp={[Function]}
        >
          Z
        </div>
        <div
          className="item-content"
        >
          <input
            className="input-component float-input"
            onBlur={[Function]}
            onChange={[Function]}
            type="text"
            value="0"
          />
        </div>
      </article>
    </div>
  </article>
  <article
    className="inspector-item wonder-three-float-input"
  >
    <div
      className="item-header"
    >
      Rotation
    </div>
    <div
      className="item-content"
    >
      <article
        className="inspector-item wonder-float-input"
      >
        <div
          className="item-header component-label"
          onMouseDown={[Function]}
          onMouseMove={[Function]}
          onMouseUp={[Function]}
        >
          X
        </div>
        <div
          className="item-content"
        >
          <input
            className="input-component float-input"
            onBlur={[Function]}
            onChange={[Function]}
            type="text"
            value="0"
          />
        </div>
      </article>
      <article
        className="inspector-item wonder-float-input"
      >
        <div
          className="item-header component-label"
          onMouseDown={[Function]}
          onMouseMove={[Function]}
          onMouseUp={[Function]}
        >
          Y
        </div>
        <div
          className="item-content"
        >
          <input
            className="input-component float-input"
            onBlur={[Function]}
            onChange={[Function]}
            type="text"
            value="0"
          />
        </div>
      </article>
      <article
        className="inspector-item wonder-float-input"
      >
        <div
          className="item-header component-label"
          onMouseDown={[Function]}
          onMouseMove={[Function]}
          onMouseUp={[Function]}
        >
          Z
        </div>
        <div
          className="item-content"
        >
          <input
            className="input-component float-input"
            onBlur={[Function]}
            onChange={[Function]}
            type="text"
            value="0"
          />
        </div>
      </article>
    </div>
  </article>
  <article
    className="inspector-item wonder-three-float-input"
  >
    <div
      className="item-header"
    >
      Scale
    </div>
    <div
      className="item-content"
    >
      <article
        className="inspector-item wonder-float-input"
      >
        <div
          className="item-header component-label"
          onMouseDown={[Function]}
          onMouseMove={[Function]}
          onMouseUp={[Function]}
        >
          X
        </div>
        <div
          className="item-content"
        >
          <input
            className="input-component float-input"
            onBlur={[Function]}
            onChange={[Function]}
            type="text"
            value="1"
          />
        </div>
      </article>
      <article
        className="inspector-item wonder-float-input"
      >
        <div
          className="item-header component-label"
          onMouseDown={[Function]}
          onMouseMove={[Function]}
          onMouseUp={[Function]}
        >
          Y
        </div>
        <div
          className="item-content"
        >
          <input
            className="input-component float-input"
            onBlur={[Function]}
            onChange={[Function]}
            type="text"
            value="1"
          />
        </div>
      </article>
      <article
        className="inspector-item wonder-float-input"
      >
        <div
          className="item-header component-label"
          onMouseDown={[Function]}
          onMouseMove={[Function]}
          onMouseUp={[Function]}
        >
          Z
        </div>
        <div
          className="item-content"
        >
          <input
            className="input-component float-input"
            onBlur={[Function]}
            onChange={[Function]}
            type="text"
            value="1"
          />
        </div>
      </article>
    </div>
  </article>
</article>
`;
